src/core/blockchain/ethereum/client-utils.ts

```diff
@@ -0,0 +1,131 @@
+import { config } from './config';
+import abi from 'ethereumjs-abi';
+import { getTransactionStatusByCode } from './transaction';
+import { generateAccountTokenState } from '../../../redux/tokens/static-selectors';
+
+export class ClientUtils implements IClientUtils {
+    constructor(private client: Client) {}
+    getTransaction(hash: string): Promise<IBlockchainTransaction> {
+        const rpcCalls = [
+            this.client.http.jsonRpc('eth_getTransactionByHash', [hash]),
+            this.client.http.jsonRpc('eth_getTransactionReceipt', [hash])
+        ];
+
+        return Promise.all(rpcCalls).then(async res => {
+            if (!res[0].result) {
+                throw new Error(
+                    res[0].error.message || `Error getting transaction info for ${hash}`
+                );
+            }
+            if (!res[1].result) {
+                throw new Error(
+                    res[1].error.message || `Error getting transaction receipt for ${hash}`
+                );
+            }
+
+            return this.buildTransactionFromBlockchain(res[0].result, res[1].result);
+        });
+    }
+
+    async buildTransactionFromBlockchain(txInfo, txReceipt) {
+        const token = await this.getToken(txInfo.to);
+        const data: any = {};
+
+        if (token.type === TokenType.ERC20) {
+            try {
+                const transferInputParameteres = this.decodeInputData(
+                    'transfer(address,uint256)',
+                    txInfo.input
+                );
+                if (transferInputParameteres[0] && transferInputParameteres[1]) {
+                    data.params = [
+                        '0x' + transferInputParameteres[0],
+                        transferInputParameteres[1].toString(10)
+                    ];
+                } else {
+                    throw new Error('Cannot decode input data');
+                }
+            } catch (e) {
+                // probably not a transaction
+                return null;
+            }
+        }
+
+        return {
+            id: txInfo.hash,
+            date: {
+                created: Date.now(),
+                signed: Date.now(),
+                broadcasted: Date.now(),
+                confirmed: Date.now()
+            },
+            blockchain: Blockchain.ETHEREUM,
+            chainId: this.client.chainId,
+            type: TransactionType.TRANSFER,
+
+            address: txInfo.from,
+            publicKey: '', // TODO: get publicKey form vrs
+
+            toAddress: txInfo.to,
+            amount: txInfo.value,
+            data,
+            feeOptions: {
+                gasPrice: txInfo.gasPrice,
+                gasLimit: txInfo.gas,
+                feeTotal: txReceipt.gasUsed
+            },
+            broadcatedOnBlock: txInfo.blockNumber,
+            nonce: txInfo.nonce,
+            status: getTransactionStatusByCode(txReceipt.status),
+            token: generateAccountTokenState(token)
```

Comment:
@georgeburduhos we should use here the token config not the token state from account...
@maricalucian we can leave it like this for now

Reply:
I'll make the changes for transactions in a separate PR